projects/packages/sync/src/sync-queue/class-queue-storage-table.php

```diff
@@ -0,0 +1,471 @@
+<?php
+/**
+ * The class responsible for storing Queue events in a dedicated Sync events table.
+ *
+ * Used by class Queue.
+ *
+ * @see \Automattic\Jetpack\Sync\Queue
+ *
+ * @package automattic/jetpack-sync
+ */
+
+namespace Automattic\Jetpack\Sync\Queue;
+
+/**
+ * Dedicated Sync events table storage backend for the Queue.
+ */
+class Queue_Storage_Table {
+	/**
+	 * The dedicated Sync events table name, without a prefix.
+	 * A prefix will be added when the class is instantiated,
+	 * as we fetch the prefix from `$wpdb` as is configured in
+	 * the WordPress config file.
+	 *
+	 * @var string
+	 */
+	public $table_name_no_prefix = 'jetpack_sync_queue';
+
+	/**
+	 * The table name with the DB prefix.
+	 *
+	 * @var string
+	 */
+	public $table_name = '';
+
+	/**
+	 * What queue is this instance responsible for.
+	 *
+	 * @var string
+	 */
+	public $queue_id = '';
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param string $queue_id The queue name this instance will be responsible for.
+	 *
+	 * @throws \Exception If queue name was not provided.
+	 */
+	public function __construct( $queue_id ) {
+		global $wpdb;
+
+		if ( empty( $queue_id ) ) {
+			// TODO what should we return here or throw an exception?
+			throw new \Exception( 'Invalid queue_id provided' );
+		}
+
+		// TODO validate the value maybe?
+		$this->queue_id = $queue_id;
+
+		// Initialize the `table_name` property with the correct prefix for easier usage in the class.
+		$this->table_name = $wpdb->prefix . $this->table_name_no_prefix;
+	}
+
+	/**
+	 * Creates the new table and updates the options to work with
+	 * the new table if it was created successfully.
+	 *
+	 * @return void
+	 */
+	public function create_table() {
+		global $wpdb;
+
+		// TODO if we run this only in the context of an upgrade, we should not include this here.
+		require_once ABSPATH . '/wp-admin/includes/upgrade.php';
+
+		$charset_collate = $wpdb->get_charset_collate();
+
+		$table_definition = "CREATE TABLE {$this->table_name} (
+			  `ID` bigint(20) NOT NULL AUTO_INCREMENT,
+			  `queue_id` varchar(50) NOT NULL,
+			  `event_id` varchar(100) NOT NULL,
+			  `event_payload` text NOT NULL,
+			  `timestamp` timestamp NOT NULL DEFAULT current_timestamp(),
+			  `object_type` varchar(100) DEFAULT NULL,
```

Comment:
Do we need the `object_type` and `parent_object_id ` here for now?

Reply:
Right, added them to test out things, but not needed for now.